problem_builder/mentoring.py

```diff
@@ -776,3 +793,98 @@ def workbench_scenarios():
+                child_content += u"<p>[{}]</p>".format(self._(u"Error: Unable to load child component."))
+            elif not isinstance(child, MentoringMessageBlock):
+                child_fragment = self._render_child_fragment(child, context, view='mentoring_view')
+                fragment.add_frag_resources(child_fragment)
+                child_content += child_fragment.content
+
+        fragment.add_content(loader.render_template('templates/html/mentoring.html', {
+            'self': self,
+            'title': self.display_name,
+            'show_title': self.show_title,
+            'child_content': child_content,
+        }))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder.css'))
+
+        self.include_theme_files(fragment)
+
+        return fragment
+
+    @property
+    def allowed_nested_blocks(self):
+        """
+        Returns a list of allowed nested XBlocks. Each item can be either
+        * An XBlock class
+        * A NestedXBlockSpec
+
+        If XBlock class is used it is assumed that this XBlock is enabled and allows multiple instances.
+        NestedXBlockSpec allows explicitly setting disabled/enabled state, disabled reason (if any) and single/multiple
+        instances
+        """
+        from .step import MentoringStepBlock  # Import here to avoid circular dependency
+        return [
+            MentoringStepBlock,
+            NestedXBlockSpec(CompletedMentoringMessageBlock, boilerplate='completed'),
+            NestedXBlockSpec(IncompleteMentoringMessageBlock, boilerplate='incomplete'),
+            NestedXBlockSpec(MaxAttemptsReachedMentoringMessageBlock, boilerplate='max_attempts_reached'),
+            NestedXBlockSpec(OnAssessmentReviewMentoringMessageBlock, boilerplate='on-assessment-review'),
+        ]
+
+    def author_edit_view(self, context):
+        """
+        Add some HTML to the author view that allows authors to add child blocks.
+        """
+        context['wrap_children'] = {
+            'head': u'<div class="mentoring">',
+            'tail': u'</div>'
+        }
+        fragment = super(MentoringWithExplicitStepsBlock, self).author_edit_view(context)
+        fragment.add_content(loader.render_template('templates/html/mentoring_url_name.html', {
+            "url_name": self.url_name
+        }))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder.css'))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder-edit.css'))
+        fragment.add_css_url(self.runtime.local_resource_url(self, 'public/css/problem-builder-tinymce-content.css'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/util.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/mentoring_with_steps_edit.js'))
+        return fragment
```

Comment:
@itsjeyd probably you've forgot to call `fragment.initialize_js('MentoringWithStepsEdit')`